server/datastore/mysql/migrations/tables/20240813103800_SupportMultipleABMTokens.go

```diff
@@ -0,0 +1,192 @@
+package tables
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240813103800, Down_20240813103800)
+}
+
+func Up_20240813103800(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE abm_tokens (
+	id                     int(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+	organization_name      varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
+	apple_id               varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
+	terms_expired          tinyint(1) NOT NULL DEFAULT '0',
+	renew_at               timestamp NOT NULL,
+	token                  blob NOT NULL,
+
+	-- those team_id fields are the default teams where devices from this ABM
+	-- will be enrolled during the DEP process, based on the device's platform
+	-- (NULL means "no team").
+	macos_default_team_id  int(10) UNSIGNED DEFAULT NULL,
+	ios_default_team_id    int(10) UNSIGNED DEFAULT NULL,
+	ipados_default_team_id int(10) UNSIGNED DEFAULT NULL,
```

Comment:
it's unclear to me from the Figma if we need to support "inactive" tokens, until they're assigned to "no team" or a team, or if they get assigned to "no team" by default.

this [dev note](https://www.figma.com/design/j2M1heOh8eZD6LcUJks6HE/%239956-Add-multiple-Apple-Business-Manager-and-Volume-Purchasing-Program-connections?node-id=5316-2969&t=TCgcHst2NZfroWrj-4) says the default value in the modal is "no team", so maybe we're fine and that's the default? otherwise we'll need to add another column, right?

Reply:
Confirmed on slack that we do want the tokens to be "active" as soon as they are added (defaulting to no team).